qiskit/algorithms/amplitude_estimators/fae.py

```diff
@@ -0,0 +1,304 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Faster Amplitude Estimation."""
+
+from typing import Optional, Union, List, Tuple
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit, ClassicalRegister
+from qiskit.providers import BaseBackend, Backend
+from qiskit.utils import QuantumInstance
+from qiskit.algorithms.exceptions import AlgorithmError
+
+from .amplitude_estimator import AmplitudeEstimator, AmplitudeEstimatorResult
+from .estimation_problem import EstimationProblem
+
+
+class FasterAmplitudeEstimation(AmplitudeEstimator):
+    """The Faster Amplitude Estimation algorithm.
+
+    The Faster Amplitude Estimation (FAE) [1] algorithm is a variant of Quantum Amplitude
+    Estimation (QAE), where the Quantum Phase Estimation (QPE) by an iterative Grover search,
+    similar to [2].
+
+    Due to the iterative version of the QPE, this algorithm does not require any additional
+    qubits, as the originally proposed QAE [3] and thus the resulting circuits are less complex.
+
+    References:
+
+        [1]: K. Nakaji. Faster Amplitude Estimation, 2020;
+            `arXiv:2002.02417 <https://arxiv.org/pdf/2003.02417.pdf>`_
+        [2]: D. Grinko et al. Iterative Amplitude Estimation, 2019;
+            `arXiv:1912.05559 <http://arxiv.org/abs/1912.05559>`_
+        [3]: G. Brassard et al. Quantum Amplitude Amplification and Estimation, 2000;
+            `arXiv:quant-ph/0005055 <http://arxiv.org/abs/quant-ph/0005055>`_
+
+    """
+
+    def __init__(self,
+                 delta: float,
+                 maxiter: int,
```

Comment:
Is there any reasonable default that could be set here? I assume this is normally a smallish number

Reply:
Yeah one could. This is essentially specifying the precision of the algorithm, like `epsilon` in the `IterativeAmplitudeEstimation` or effectively `num_evaluation_qubits` in `AmplitudeEstimation`. Since these are not set by default I'd rather keep this as is -- or add defaults everywhere.